src/throttler.service.ts

```diff
@@ -8,26 +8,33 @@ import { ThrottlerStorage } from './throttler-storage.interface';
  */
 @Injectable()
 export class ThrottlerStorageService implements ThrottlerStorage, OnApplicationShutdown {
-  private _storage: Record<string, ThrottlerStorageOptions> = {};
+  private _map: Map<string, ThrottlerStorageOptions> = new Map();
   private timeoutIds: NodeJS.Timeout[] = [];
 
   get storage(): Record<string, ThrottlerStorageOptions> {
-    return this._storage;
+    return Object.fromEntries(this.map);
+    // If need to increase performance with loss of access to Map data.
+    // Detailed behavior: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#setting_object_properties
+    // return this._map as unknown as Record<string, ThrottlerStorageOptions>;
+  }
+
+  private get map(): Map<string, ThrottlerStorageOptions> {
+    return this._map;
   }
```

Comment:
we can drop this method in favor of accessing `this._map` directly inside this class